diff -urN gnome-vfs-2.6.0.orig/configure.in gnome-vfs-2.6.0/configure.in
--- gnome-vfs-2.6.0.orig/configure.in	2004-03-22 17:43:00.606033680 +0100
+++ gnome-vfs-2.6.0/configure.in	2004-03-22 17:43:38.332298416 +0100
@@ -127,7 +127,7 @@
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE")
 
-ALL_LINGUAS="am ar az be bg bn bs ca cs cy da de el en_GB eo es et eu fa fi fr ga gl he hi hr hu id is it ja ko li lt lv mk ml mn mr ms ne nl nn no pa pl pt pt_BR ro ru sk sl sq sr sr@ije sr@Latn sv ta th tr uk vi wa yi zh_CN zh_TW"
+ALL_LINGUAS="am ar az be bg bn bs ca cs cy da de el en_GB eo es et eu fa fi fr ga gl he hi hr hu id is it ja ko li lt lv mk ml mn mr ms nb ne nl nn pa pl pt pt_BR ro ru sk sl sq sr sr@ije sr@Latn sv ta th tr uk vi wa yi zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 
