diff -urN gnome-vfs-2.0.3.orig/modules/vfolder/applications-all-users.vfolder-info.in gnome-vfs-2.0.3/modules/vfolder/applications-all-users.vfolder-info.in
--- gnome-vfs-2.0.3.orig/modules/vfolder/applications-all-users.vfolder-info.in	Thu Aug 29 13:19:36 2002
+++ gnome-vfs-2.0.3/modules/vfolder/applications-all-users.vfolder-info.in	Thu Aug 29 13:21:53 2002
@@ -17,6 +17,7 @@
   <!-- Merge in the GNOME1 menus if you build into a non-default prefix -->
   <MergeDir>@datadir@/gnome/apps/</MergeDir>
   <MergeDir>/usr/share/gnome/apps/</MergeDir>
+  <MergeDir>/usr/X11R6/share/applnk/</MergeDir>
 
   <!-- Merge dirs optional -->
   <MergeDir>/etc/X11/applnk/</MergeDir>
