--- gnome-vfs-2.3.2/configure.in.orig	Mon May 19 10:56:15 2003
+++ gnome-vfs-2.3.2/configure.in	Tue May 20 11:34:49 2003
@@ -125,7 +125,7 @@
 AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE")
 
-ALL_LINGUAS="am ar az be bg bn bs ca cs cy da de el eo es et eu fa fi fr ga gl he hi hu id is it ja ko lt lv mk ml mn ms nl nn no pl pt pt_BR ro ru sk sl sq sr sr@Latn sv tr uk vi wa yi zh_CN zh_TW"
+ALL_LINGUAS="am ar az be bg bn bs ca cs cy da de el eo es et eu fa fi fr ga gl he hi hu id is it ja ko lt lv mk ml mn ms nl nn no pl pt pt_BR ro ru sk sl sq sr sr@cyrillic sv tr uk vi wa yi zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 AC_FUNC_ALLOCA
