diff -urN gnome-vfs-2.11.92.orig/libgnomevfs/xdgmime.c gnome-vfs-2.11.92/libgnomevfs/xdgmime.c
--- gnome-vfs-2.11.92.orig/libgnomevfs/xdgmime.c	2005-08-24 14:42:16.910606480 +0200
+++ gnome-vfs-2.11.92/libgnomevfs/xdgmime.c	2005-08-24 14:42:50.795455200 +0200
@@ -234,7 +234,7 @@
 
   xdg_data_dirs = getenv ("XDG_DATA_DIRS");
   if (xdg_data_dirs == NULL)
-    xdg_data_dirs = "/usr/local/share/:/usr/share/";
+    xdg_data_dirs = "/usr/local/share/:/usr/share/gnome/:/usr/share/";
 
   ptr = xdg_data_dirs;
 
