diff -Nuard gnome-vfs-2.4.1.orig/modules/vfolder/applications-all-users.vfolder-info.in gnome-vfs-2.4.1/modules/vfolder/applications-all-users.vfolder-info.in
--- gnome-vfs-2.4.1.orig/modules/vfolder/applications-all-users.vfolder-info.in	2003-09-27 17:42:52.000000000 +0200
+++ gnome-vfs-2.4.1/modules/vfolder/applications-all-users.vfolder-info.in	2003-12-14 19:34:10.216409656 +0100
@@ -199,6 +199,9 @@
           <Not>
             <Keyword>Settings</Keyword>
           </Not>
+          <Not>
+            <Keyword>KDE</Keyword>
+          </Not>
           <Keyword>System</Keyword>
         </And>
       </Query>
diff -Nuard gnome-vfs-2.4.0.orig/modules/vfolder/applications-all-users.vfolder-info.in gnome-vfs-2.4.0/modules/vfolder/applications-all-users.vfolder-info.in
--- gnome-vfs-2.4.0.orig/modules/vfolder/applications-all-users.vfolder-info.in	2003-09-20 00:46:23.000000000 +0200
+++ gnome-vfs-2.4.0/modules/vfolder/applications-all-users.vfolder-info.in	2003-09-20 00:54:23.000000000 +0200
@@ -47,7 +47,6 @@
       <Desktop>Accessories.directory</Desktop>
       <Query>
 	<And>
-	  <Keyword>Application</Keyword>
 	  <Keyword>Utility</Keyword>
 	  <Not>
 	    <Keyword>System</Keyword>
@@ -63,7 +62,6 @@
       <Desktop>Accessibility.directory</Desktop>
       <Query>
 	<And>
-	  <Keyword>Application</Keyword>
 	  <Keyword>Accessibility</Keyword>
 	  <Not>
 	    <Keyword>Settings</Keyword>
@@ -80,7 +78,6 @@
       <OnlyUnallocated/>
       <Query>
 	<And>
-          <Keyword>Application</Keyword>
           <Not>
             <Keyword>Settings</Keyword>
           </Not>
@@ -101,7 +98,6 @@
       <Desktop>Development.directory</Desktop>
       <Query>
         <And>
-          <Keyword>Application</Keyword>
 	  <Keyword>Development</Keyword>
         </And>
       </Query>
@@ -114,7 +110,6 @@
       <Desktop>Games.directory</Desktop>
       <Query>
         <And>
-          <Keyword>Application</Keyword>
 	  <Keyword>Game</Keyword>
         </And>
       </Query>
@@ -127,7 +122,6 @@
       <Desktop>Graphics.directory</Desktop>
       <Query>
         <And>
-          <Keyword>Application</Keyword>
 	  <Keyword>Graphics</Keyword>
         </And>
       </Query>
@@ -140,7 +134,6 @@
       <Desktop>Internet.directory</Desktop>
       <Query>
         <And>
-          <Keyword>Application</Keyword>
 	  <Keyword>Network</Keyword>
         </And>
       </Query>
@@ -153,7 +146,6 @@
       <Desktop>Multimedia.directory</Desktop>
       <Query>
         <And>
-          <Keyword>Application</Keyword>
 	  <Keyword>AudioVideo</Keyword>
         </And>
       </Query>
@@ -170,7 +162,6 @@
       <Desktop>Office.directory</Desktop>
       <Query>
 	<And>
-	  <Keyword>Application</Keyword>
 	  <Or>
 	    <Keyword>Office</Keyword>
 	    <Keyword>Spreadsheet</Keyword>
@@ -195,7 +186,6 @@
       <Desktop>System.directory</Desktop>
       <Query>
         <And>
-          <Keyword>Application</Keyword>
           <Not>
             <Keyword>Settings</Keyword>
           </Not>
