--- gnome-vfs-1.9.15/monikers/Makefile.am~	Tue May 14 21:37:53 2002
+++ gnome-vfs-1.9.15/monikers/Makefile.am	Tue May 14 21:37:53 2002
@@ -17,7 +17,7 @@
 		$(srcdir)/GNOME_VFS_Moniker_std.server.in.in \
 		> GNOME_VFS_Moniker_std.server.in
 
-CLEANFILES+=GNOME_VFS_Moniker_std.server GNOME_VFS_Moniker_std.server.in
+CLEANFILES = GNOME_VFS_Moniker_std.server GNOME_VFS_Moniker_std.server.in
 
 serverdir = $(libdir)/bonobo/servers
 server_DATA = GNOME_VFS_Moniker_std.server
