diff -urN aa/gnome-vfs-2.5.6/daemon/Makefile.am gnome-vfs-2.5.6/daemon/Makefile.am
--- aa/gnome-vfs-2.5.6/daemon/Makefile.am	2004-01-20 09:41:57.000000000 +0100
+++ gnome-vfs-2.5.6/daemon/Makefile.am	2004-01-30 23:30:44.238831080 +0100
@@ -50,8 +50,8 @@
 serverfile = GNOME_VFS_Daemon.server.in
 serverdir = $(libdir)/bonobo/servers
 install-data-hook :
-	$(mkinstalldirs) $(serverdir)
-	sed -e "s|\@LIBEXECDIR\@|$(libexecdir)|" $(srcdir)/$(serverfile) > $(serverdir)/$(serverfile:.in=)
+	$(mkinstalldirs) $(DESTDIR)$(serverdir)
+	sed -e "s|\@LIBEXECDIR\@|$(libexecdir)|" $(srcdir)/$(serverfile) > $(DESTDIR)$(serverdir)/$(serverfile:.in=)
 
 
 EXTRA_DIST = $(serverfile)
