diff -Nuard gnome-vfs-2.4.1.orig/modules/vfolder/vfolder-common.c gnome-vfs-2.4.1/modules/vfolder/vfolder-common.c
--- gnome-vfs-2.4.1.orig/modules/vfolder/vfolder-common.c	2003-09-27 17:42:52.000000000 +0200
+++ gnome-vfs-2.4.1/modules/vfolder/vfolder-common.c	2003-12-14 18:52:30.369444160 +0100
@@ -201,13 +201,33 @@
 entry_reload_if_needed (Entry *entry)
 {
 	gboolean changed = FALSE;
-	gchar *keywords, *deprecates;
+	gchar *keywords, *deprecates, *onlyshowin;
 	int i;
 
 	if (!entry->dirty)
 		return;
 
 	entry_quick_read_keys (entry, 
+			       "OnlyShowIn",
+			       &onlyshowin,
+			       NULL,
+			       NULL);
+
+
+	if (onlyshowin) {
+		char **parsed = g_strsplit (onlyshowin, ";", -1);
+		gboolean show = FALSE;
+		
+		for (i = 0; parsed[i] != NULL; i++) {
+			if (!g_ascii_strcasecmp(parsed[i],"GNOME"))
+				show=TRUE;
+		}
+		
+		g_free (onlyshowin);
+		if (!show) return;
+	}
+
+	entry_quick_read_keys (entry, 
 			       "Categories",
 			       &keywords,
 			       "Deprecates",
