diff -urN aa/gnome-vfs-2.7.90/libgnomevfs/gnome-vfs-dns-sd.h gnome-vfs-2.7.90/libgnomevfs/gnome-vfs-dns-sd.h
--- aa/gnome-vfs-2.7.90/libgnomevfs/gnome-vfs-dns-sd.h	2004-07-31 13:27:44.000000000 +0200
+++ gnome-vfs-2.7.90/libgnomevfs/gnome-vfs-dns-sd.h	2004-08-08 21:47:04.679667080 +0200
@@ -47,7 +47,7 @@
 
 typedef enum {
 	GNOME_VFS_DNS_SD_SERVICE_ADDED,
-	GNOME_VFS_DNS_SD_SERVICE_REMOVED,
+	GNOME_VFS_DNS_SD_SERVICE_REMOVED
 } GnomeVFSDNSSDServiceStatus;
 
 GType gnome_vfs_dns_sd_service_status_get_type (void);
