diff -urN gnome-vfs-2.1.3.1.orig/libgnomevfs/Makefile.am gnome-vfs-2.1.3.1/libgnomevfs/Makefile.am
--- gnome-vfs-2.1.3.1.orig/libgnomevfs/Makefile.am	Thu Nov 28 17:20:10 2002
+++ gnome-vfs-2.1.3.1/libgnomevfs/Makefile.am	Thu Nov 28 17:23:43 2002
@@ -63,7 +63,6 @@
 	gnome-vfs-ssl-private.h			\
 	gnome-vfs-thread-pool.h			\
 	$(NULL)
-#	gnome-vfs-metadata-private.h		\
 
 libgnomevfsmoduleinclude_HEADERS =		\
 	gnome-vfs-cancellable-ops.h		\
@@ -108,7 +107,6 @@
 	gnome-vfs-xfer.h			\
 	gnome-vfs.h				\
 	$(NULL)
-#	gnome-vfs-metadata.h			\
 
 libgnomevfs_2_la_SOURCES =			\
 	gnome-vfs-application-registry.c	\
