--- gnome-vfs-1.9.17/modules/applications.vfolder-info.in.wiget	Mon Jun 10 14:42:49 2002
+++ gnome-vfs-1.9.17/modules/applications.vfolder-info.in	Mon Jun 10 14:43:32 2002
@@ -5,6 +5,7 @@
 
   <!-- Merges in the main GNOME menus if you build into a non-default prefix -->
   <MergeDir>/usr/share/gnome/apps/</MergeDir>
+  <MergeDir>/usr/X11R6/share/applnk/</MergeDir>
 
   <!-- Control Center Merge Directories -->
   <MergeDir>@datadir@/control-center-2.0/capplets/</MergeDir>
