diff -Nur gnome-vfs2-2.14.0/libgnomevfs/gnome-vfs-parse-ls.c gnome-vfs2-2.14.0.new/libgnomevfs/gnome-vfs-parse-ls.c
--- gnome-vfs2-2.14.0/libgnomevfs/gnome-vfs-parse-ls.c	2005-11-14 12:24:14.000000000 +0100
+++ gnome-vfs2-2.14.0.new/libgnomevfs/gnome-vfs-parse-ls.c	2006-04-18 21:35:02.000000000 +0200
@@ -624,7 +624,7 @@
 		if (!is_num (columns[idx2]))
 			goto error;
 	
-		s->st_size = (gsize) atol (columns [idx2]);
+		s->st_size = (gsize) atoll (columns [idx2]);
 #ifdef HAVE_ST_RDEV
 		s->st_rdev = 0;
 #endif
